dev/com.ibm.ws.persistence/src/com/ibm/wsspi/persistence/internal/DatabaseStoreImpl.java

```diff
@@ -192,19 +200,54 @@ public PersistenceServiceUnit createPersistenceServiceUnit(ClassLoader loader, S
         }
 
         // Look for EclipseLink persistence properties
-        if (properties.get("persistenceProperties.0.config.referenceType") != null) {
-            for (String key : Collections.list(properties.keys())) {
+        if (this.properties.get("persistenceProperties.0.config.referenceType") != null) {
+            for (String key : Collections.list(this.properties.keys())) {
                 if (key.startsWith("persistenceProperties.0") && !key.equals("persistenceProperties.0.config.referenceType")) {
-                    Object value = properties.get(key);
+                    Object value = this.properties.get(key);
                     key = key.substring(24);
                     puProps.put(key, value);
                 }
             }
         }
 
+        // Look for additional configuration properties
```

Comment:
This is a big chunk of specialized code to add to this method, I wonder if the code involved in determining the isolation level should be moved to its own method, and we just use the value returned by that method in resourceInfo.setIsolationLevel().  This way it's less of a change to the original method body.

Reply:
Sure, I'll move to a sub-method